docs/source/discord/resources/channel/methods/get_pinned_messages.rst

```diff
@@ -8,12 +8,19 @@ Example
 
 .. code:: c
 
-   struct discord_message **ret;
-   
-   discord_get_pinned_messages(client, msg->channel_id, &ret);
-   
-   if (ret)
-     for (int i = 0; ret[i]; ++i)
-       printf("%s\n", ret[i]->content);
-       
-   discord_message_list_free(ret);
+   void done_get_pinned_messages(struct discord *client, void *data, const struct discord_messages *ret)
+   {
+     if (ret)
+       for (int i = 0; ret[i]; ++i)
+         printf("%s\n", ret[i]->content);
```

Comment:
Please check the [guide](https://gist.github.com/lcsmuller/a5f2b205c3871888656b86825db90187#example-3) for the new method of iterating our built-in array types.